### DISCUSSION TITLE
What's the recommended way to make a snippet fire only inside an environment?

### DISCUSSION THREAD

**Discussion by medwatt** [L3MON4D3/LuaSnip#955] [Q&A]
The snippet below creates a snippet for the `if` statement. Suppose I would like to bind the keys `@@` to create an `elseif` block under the condition that `@@` only does this when I'm inside an `if` statement. How can I do that?

```lua
s(
    { trig = "if", dscr = "if statement" },
    fmta(
        [[
            if <> then
                <>
            end
        ]],
        {
            i(1, "condition"),

            c(2,
                {
                    i(1),
                    sn(2, { i(1), t({"","else", "\t"}), i(2) }),
                    sn(3, { i(1), t({"","elseif "}), i(2, "condition"), t({" then", "\t"}), i(3), t({"","else", "\t"}), i(4) }),
                }
            )
        }
    ),
)
```

**Comment by L3MON4D3** [comment]
Check out `:h luasnip-snippets` and look for `condition`, it makes it possible to disallow expansion if some condition (cursor in an if-statement) does not hold

**Reply by medwatt** [reply]
Thanks for the reply. There's a lot of `condition` match. So, I don't know which part you're referring to.

As an additional question, how does the snippet know when the cursor is inside an if statement?

**Reply by L3MON4D3** [reply]
> There's a lot of `condition` match. So, I don't know which part you're referring to.

I can't link to the exact line , unfortunately, but there's a list of valid keys for the first argument to `snippet`/`s`, it's in there (search for "but we already have that info here", that should be unique)

> As an additional question, how does the snippet know when the cursor is inside an if statement?

Ah, that's for you to figure out :sweat_smile:
treesitter is pretty good at stuff like this, but if you're not familiar with that it might take a bit to get it working.. just searching the buffer is also an option, but more prone to errors.
